opm/simulators/wells/BlackoilWellModelConstraints.cpp

```diff
@@ -412,9 +415,20 @@ actionOnBrokenConstraints(const Group& group,
     std::string ss;
     switch(exceed_action) {
     case Group::ExceedAction::NONE: {
+        // phase specific exceed actions need to be considered
         if (oldControl != newControl && oldControl != Group::ProductionCMode::NONE) {
-            ss = fmt::format("Group production exceed action is NONE for group {}. Nothing happens.",
-                             group.name());
+            if ((water_exceed_action == Group::ExceedAction::RATE && newControl == Group::ProductionCMode::WRAT) ||
+                (gas_exceed_action == Group::ExceedAction::RATE && newControl == Group::ProductionCMode::GRAT) ||
+                (liquid_exceed_action == Group::ExceedAction::RATE && newControl == Group::ProductionCMode::LRAT)) {
+                group_state.production_control(group.name(), newControl);
+                ss = fmt::format("Switching production control mode for group {} from {} to {}",
+                                 group.name(),
+                                 Group::ProductionCMode2String(oldControl),
+                                 Group::ProductionCMode2String(newControl));
+            } 
+            else {
+                ss = fmt::format("Group production exceed action is NONE for group {}. Nothing happens.", group.name());
```

Comment:
This message does not seem clear enough to me. How about something like: "Procedure on exceeding the limit is NONE for group {}. Nothing is done."